core/src/main/kotlin/io/specmatic/core/pattern/JSONObjectPattern.kt

```diff
@@ -398,7 +408,45 @@ data class JSONObjectPattern(
         }
     }
 
+    private fun resolveJSONObjectExample(example: Any?, pattern: JSONObjectPattern, resolver: Resolver): JSONObjectValue? {
+        if (example == null) return null
+
+        val value = when (example) {
+            is String -> pattern.parse(example, resolver)
```

Comment:
Don't pattern.parse an example. If it's a string, so be it. Also in addition to map, handle the other cases, namely if value is a number, boolean, null, array.